Bottom: 178c7f657abdc27983f53804fd87eb3139029fa8
Top:    a2686301246303828c867b2d2423a1b889794717
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 23:09:45 +0100

Add a HID API call, notify_save_pcb() called around saving the PCB

The intention of this API is so that GUIs monitoring the active PCB
file on disk for changes, can filter out changes which occur as we
save the file ourselves.


---

diff --git a/src/file.c b/src/file.c
index 3fdc93d..be7f6a5 100644
--- a/src/file.c
+++ b/src/file.c
@@ -349,9 +349,18 @@ SaveBufferElements (char *Filename)
  * save PCB
  */
 int
-SavePCB (char *Filename)
+SavePCB (char *file)
 {
-  return WritePipe (Filename, true);
+  int retcode;
+
+  if (gui->notify_save_pcb == NULL)
+    return WritePipe (file, true);
+
+  gui->notify_save_pcb (file, false);
+  retcode = WritePipe (file, true);
+  gui->notify_save_pcb (file, true);
+
+  return retcode;
 }
 
 /* ---------------------------------------------------------------------------
diff --git a/src/hid.h b/src/hid.h
index 322602a..8b39eda 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -568,6 +568,16 @@ typedef enum
      */
     void (*finish_debug_draw)  (void);
 
+    /* Notification to the GUI around saving the PCB file.
+     *
+     * Called with a false parameter before the save, called again
+     * with true after the save.
+     *
+     * Allows GUIs which watch for file-changes on disk to ignore
+     * our deliberate changes.
+     */
+    void (*notify_save_pcb) (const char *filename, bool done);
+
     /* Notification to the GUI that the PCB file is being renamed. */
     void (*notify_pcb_filename_change) (const char *old_filename,
                                         const char *new_filename);
